tests/run-pass/intptrcast-truncate.rs

```diff
@@ -0,0 +1,9 @@
+// compile-flags: -Zmiri-seed=0000000000000000
+
+fn main() {
+    let x = &42 as *const i32 as usize; 
+    let y = x * 2;
+    assert_eq!(y, x + x);
+    let z = y as u8 as usize;
+    assert_eq!(z, y % 256);
+}
```

Comment:
The filename for this is a bit odd now. And anway we usually avoid having just a single thing in a run-pass test, they are too slow for that. So maybe just call the file `intptrcast`? We can extend it with more tests as time goes by.